src/journal/journal-file.c

```diff
@@ -86,7 +88,82 @@
 /* The mmap context to use for the header we pick as one above the last defined typed */
 #define CONTEXT_HEADER _OBJECT_TYPE_MAX
 
-static int journal_file_set_online(JournalFile *f) {
+static int journal_file_set_offline_fd(int fd) {
+        typeof(((Header *)0)->state) state = STATE_OFFLINE;
+        int r;
+
+        assert_cc(sizeof(state) == 1);
+        assert(fd >= 0);
+
+        r = pwrite(fd, &state, sizeof(state), offsetof(Header, state));
+        if (r < 0)
+                return r;
+
+        r = fsync(fd);
+        if (r < 0)
+                return r;
+
+        return r;
+}
+
+
+static int journal_file_set_offline_internal(JournalFile *f) {
```

Comment:
@poettering Do you feel this is actually necessary with how I have things implemented here?  I don't think it is, since the header is a keep_always mmap, can't we safely just write to f->header->state in the offline thread during the OFFLINE_OFFLINING phase and drop this function altogether?